apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -0,0 +1,188 @@
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+
+	"net/http"
+
+	"github.com/vmware/vic/apiservers/portlayer/models"
+	"github.com/vmware/vic/apiservers/portlayer/restapi/operations"
+	"github.com/vmware/vic/apiservers/portlayer/restapi/operations/containers"
+	"github.com/vmware/vic/apiservers/portlayer/restapi/options"
+	"github.com/vmware/vic/metadata"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/portlayer/exec"
+)
+
+// ContainersHandlersImpl is the receiver for all of the exec handler methods
+type ContainersHandlersImpl struct {
+	handlerCtx *HandlerContext
+}
+
+const (
+	serialOverLANPort = 2377
+)
+
+// Configure assigns functions to all the exec api handlers
+func (handler *ContainersHandlersImpl) Configure(api *operations.PortLayerAPI, handlerCtx *HandlerContext) {
+	api.ContainersCreateHandler = containers.CreateHandlerFunc(handler.CreateHandler)
+	api.ContainersStateChangeHandler = containers.StateChangeHandlerFunc(handler.StateChangeHandler)
+	api.ContainersGetHandler = containers.GetHandlerFunc(handler.GetHandler)
+	api.ContainersCommitHandler = containers.CommitHandlerFunc(handler.CommitHandler)
+
+	handler.handlerCtx = handlerCtx
+}
+
+// CreateHandler creates a new container
+func (handler *ContainersHandlersImpl) CreateHandler(params containers.CreateParams) middleware.Responder {
+	defer trace.End(trace.Begin("Containers.CreateHandler"))
+
+	var err error
+	var name string
+	session := handler.handlerCtx.Session
+
+	ctx := context.Background()
+
+	log.Debugf("Path: %#v", params.CreateConfig.Path)
+	log.Debugf("Args: %#v", params.CreateConfig.Args)
+	log.Debugf("Env: %#v", params.CreateConfig.Env)
+	log.Debugf("WorkingDir: %#v", params.CreateConfig.WorkingDir)
+
+	id := exec.GenerateID().String()
+	// Autogenerate a name if client doesn't specify one
+	if params.Name == nil {
+		name = namesgenerator.GetRandomName(0)
+	} else {
+		name = *params.Name
+	}
+
+	// Init key for tether
+	privateKey, err := rsa.GenerateKey(rand.Reader, 512)
+	if err != nil {
+		return containers.NewCreateNotFound().WithPayload(&models.Error{Message: err.Error()})
+	}
+	privateKeyBlock := pem.Block{
+		Type:    "RSA PRIVATE KEY",
+		Headers: nil,
+		Bytes:   x509.MarshalPKCS1PrivateKey(privateKey),
+	}
+
+	m := metadata.ExecutorConfig{
+		Common: metadata.Common{
+			ID:   id,
+			Name: name,
+		},
+		Sessions: map[string]metadata.SessionConfig{
+			id: metadata.SessionConfig{
+				Common: metadata.Common{
+					ID: id,
+				},
+				Tty: false,
+				// FIXME: default to true for now until we can have a more sophisticated approach
+				Attach: true,
+				Cmd: metadata.Cmd{
+					Env:  params.CreateConfig.Env,
+					Dir:  *params.CreateConfig.WorkingDir,
+					Path: *params.CreateConfig.Path,
+					Args: append([]string{*params.CreateConfig.Path}, params.CreateConfig.Args...),
+				},
+			},
+		},
+		Networks: make(map[string]*metadata.NetworkEndpoint),
```

Comment:
Might be neater to add this in the scopes.AddContainer impl - there's no guarantee we have any networks to put in it at this point